atc/atccmd/command.go

```diff
@@ -214,15 +214,16 @@ type RunCommand struct {
 	DefaultMemoryLimit *string `long:"default-task-memory-limit" description:"Default maximum memory per task, 0 means unlimited"`
 
 	Auditor struct {
-		EnableBuildAuditLog     bool `long:"enable-build-auditing" description:"Enable auditing for all api requests connected to builds."`
-		EnableContainerAuditLog bool `long:"enable-container-auditing" description:"Enable auditing for all api requests connected to containers."`
-		EnableJobAuditLog       bool `long:"enable-job-auditing" description:"Enable auditing for all api requests connected to jobs."`
-		EnablePipelineAuditLog  bool `long:"enable-pipeline-auditing" description:"Enable auditing for all api requests connected to pipelines."`
-		EnableResourceAuditLog  bool `long:"enable-resource-auditing" description:"Enable auditing for all api requests connected to resources."`
-		EnableSystemAuditLog    bool `long:"enable-system-auditing" description:"Enable auditing for all api requests connected to system transactions."`
-		EnableTeamAuditLog      bool `long:"enable-team-auditing" description:"Enable auditing for all api requests connected to teams."`
-		EnableWorkerAuditLog    bool `long:"enable-worker-auditing" description:"Enable auditing for all api requests connected to workers."`
-		EnableVolumeAuditLog    bool `long:"enable-volume-auditing" description:"Enable auditing for all api requests connected to volumes."`
+		EnableBuildAuditLog     bool   `long:"enable-build-auditing" description:"Enable auditing for all api requests connected to builds."`
+		EnableContainerAuditLog bool   `long:"enable-container-auditing" description:"Enable auditing for all api requests connected to containers."`
+		EnableJobAuditLog       bool   `long:"enable-job-auditing" description:"Enable auditing for all api requests connected to jobs."`
+		EnablePipelineAuditLog  bool   `long:"enable-pipeline-auditing" description:"Enable auditing for all api requests connected to pipelines."`
+		EnableResourceAuditLog  bool   `long:"enable-resource-auditing" description:"Enable auditing for all api requests connected to resources."`
+		EnableSystemAuditLog    bool   `long:"enable-system-auditing" description:"Enable auditing for all api requests connected to system transactions."`
+		EnableTeamAuditLog      bool   `long:"enable-team-auditing" description:"Enable auditing for all api requests connected to teams."`
+		EnableWorkerAuditLog    bool   `long:"enable-worker-auditing" description:"Enable auditing for all api requests connected to workers."`
+		EnableVolumeAuditLog    bool   `long:"enable-volume-auditing" description:"Enable auditing for all api requests connected to volumes."`
+		ClientIPHeader          string `long:"client-ip-header" description:"A header that a reverse proxy will set to the client's IP address. If not set then X-REAL-IP and X-FORWARD-FOR will be used in that order."`
```

Comment:
I was thinking more along the lines of just adding some header(s), not necessarily for the IP address, that we just pass along as is and (don't try to infer any meaning from). In addition to logging the custom header(s), we'd log `r.RemoteAddr` as is.

So, if the operator configured logging the header `X-Forwarded-For`, then we'd log something like:

```
{"action": "...", "user": "...", "ip": "<remote proxy IP>", headers: {"X-Forwarded-For": "<local ip, possibly a chain, possibly spoofed>"}}
```

Or, if the operator configured logging the header `X-Request-Id`:

```
{"action": "...", "user": "...", "ip": "<remote proxy IP>", headers: {"X-Request-Id": "<some id that can be traced through the remote proxy logs>"}}
```

What do you think about that? I think always logging `r.RemoteAddr` is a good idea, since we can't possibly know what configuration of proxies is sitting in front of Concourse, if any. Allowing them to log any header(s) also seems like a good idea, since it allows you to log things like `X-Request-Id`, which can't be spoofed

Reply:
> Allowing them to log any header(s) also seems like a good idea, since it allows you to log things like X-Request-Id, which can't be spoofed (if you've configured your reverse proxy correctly).

and (just to add to the "it's hard to get a meaningful ip"), nowadays it's just sooooo easy to use any of those browsers w/ VPN support built-in, or even [tor](https://www.torproject.org/) (one can literally just go to a private tab on brave and enable it or download the tor browser)

Reply:
@taylorsilva just pinging you about this - any thoughts about this approach?